### ISSUE TITLE
Option to ignore order of array elements

### ISSUE THREAD

**Issue by rrva** [flipkart-incubator/zjsonpatch#32]
I am looking for a way to ignore the order of array elements, so that 

```json
{
  "foo": [0,1,2]
}
```

is equal to

```json
{
  "foo": [0,2,1]
}
```

**Comment by vishwakarma** [comment]
Hi @rrva 
Currently as per the RFC, 2 arrays are considered as not equal even if they have elements in different order.
Hence, currently there is no support as requested by you.

**Comment by rrva** [comment]
Yes, I was asking for an _optional_ way (not on by default to accomplish this)

Currently I was thinking to postprocess the diff using something like this:

```java
```

**Comment by vishwakarma** [comment]
Yeah, currently there is not any provision for optionally ignoring diffs in ordering of elements in array. You need to work on the diffs generated by  excluding move operations within array.

**Comment by rrva** [comment]
Ok, closing for now then if what I pasted is the best way to currently do it, and there are no plans to make it configurable.

**Comment by naumanshah03** [comment]
@rrva Can you help me to achieve this feature?
What changes would be required and how to use it?

**Comment by holograph** [comment]
This would essentially require treating all arrays as sets. You could probably achieve this by rewriting `JsonDiff.compareArray` to create a set of array elements on both source and destination, compare and apply accordingly; this really goes against the grain of the RFC, though, and the library was never designed for it, so there might be more work to do.

**Comment by naumanshah03** [comment]
My current approach,
Perform a JSON sort (JSON Object cannot be sorted so provided some parameter based on which this sort will happen) and then give it for diff. This has solved my main concern of Array reorder (although it reorder is considered a diff)

Thank you for your response. 
I have checked through the open and closed issues and I saw that this is the most heavily requested issue even though it is not compliant with the RFC. I thought that there might be someone with this implementation.

@holograph I really appreciate the library and the people working on it. I will surely look into the implementation
